### PR TITLE
Poll old transactions

### LINKED ISSUE #24: Minimize chances of missed withdrawals
**Is your feature request related to a problem? Please describe.**
Right now we watch for transfer events using the subscribe rpc feature, we should more robustly handle any instances where we might miss transfers

**Describe the solution you'd like**
In addition to eth_subscribe, consider one or both of:

- Periodic polling of event logs since last polled block, ensuring that all transfers that meet confirmation requirements were processed
- A way to manually trigger processing a transaction by providing a txhash, by which we could retrieve the event logs and check reqs / ensure processed

**Describe alternatives you've considered**
In order to miss a transfer >1/3 of verifiers would have to miss it, we could increase number of verifiers to decrease this probability with no code changes.

### PR DESCRIPTION
**Pull request by Rizato**
Fixes #24

### CODE REVIEW

File: src/relay.rs [hunk 1]
```diff
@@ -293,6 +376,144 @@ impl<T: DuplexTransport + 'static> Network<T> {
...
+        let h = handle.clone();
+        let token_address = self.token.address().clone();
+        let relay_address = self.relay.address().clone();
+        let network_type = self.network_type;
+        let web3 = self.web3.clone();
+        let future = interval.for_each(move |_| {
+            let handle = h.clone();
+            let tx = tx.clone();
+            let web3 = web3.clone();
+            let w3 = web3.clone();
```
Comment (mjkoo) [inline src/relay.rs]: I get why `h` and `w3` exist but i think there's probably a cleaner way to work around the borrow checker here, gonna run out of variable names at that rate.

File: src/relay.rs [hunk 2]
```diff
@@ -293,6 +376,144 @@ impl<T: DuplexTransport + 'static> Network<T> {
...
+                    let web3 = web3.clone();
+                    web3.eth().logs(filter.clone())
+                }).and_then(move |logs| {
+                    let tx = tx.clone();
+                    let web3 = w3.clone();
+                    let handle = handle.clone();
+                    info!("found {} log(s)", &logs.len());
+                    logs.iter().for_each(|log| {
+                        if Some(true) == log.removed {
+                            info!("removed logs :(");
```
Comment (mjkoo) [inline src/relay.rs]: Probably should be a warn! (might reconsider message too), and also we don't want to break early if one log is removed right?

File: src/relay.rs [hunk 3]
```diff
@@ -293,6 +376,144 @@ impl<T: DuplexTransport + 'static> Network<T> {
...
+                            return;
+                        }
+
+                        log.transaction_hash.map_or_else(|| {
+                            warn!("log missing transaction hash");
+                            return;
```
Comment (mjkoo) [inline src/relay.rs]: Same as above, we should continue processing logs i think right?

File: src/relay.rs [hunk 4]
```diff
@@ -293,6 +386,157 @@ impl<T: DuplexTransport + 'static> Network<T> {
...
+                Err(())
+            })
+    }
+
+    /// Returns a Stream of Transfer that were missed, either from downtime, or failed approvals
+    ///
+    /// # Arguments
+    ///
+    /// * `handle` - Handle to the event loop to spawn additional futures
+    pub fn missed_transfer_stream(
```
Comment (zv) [inline src/relay.rs]: Add `interval` to the docstring above.

**Pull request merged**